lib/build.js

```diff
@@ -339,13 +342,76 @@ const parseAct = (act_str, silent = false) => {
 }
 
 
+const buildAct = config => ({act, oog, pass, name}) => {
+  console.log("build " + name)
 
+  // HELPERS
+  // TODO - sideeffect free
+  const getTerm = ({
+    output,
+    statusCode,
+    accounts,
+    gas,
+    pc
+  }) => ({
+      "k": "#execute ~> CONTINUATION => " + ((act.internal && pass && !(oog)) ? "#execute ~> CONTINUATION" : "#halt ~> CONTINUATION"),
+    "ethereum.evm.callState.programBytes": `#parseByteStack("0x${config.contracts[act.subject].bin_runtime}")`,
+    "ethereum.evm.callState.program":  `#asMapOpCodes(#dasmOpCodes(#parseByteStack("0x${config.contracts[act.subject].bin_runtime}"), BYZANTIUM))`,
+    "ethereum.evm.callState.callData": act.callData,
+    "ethereum.evm.callState.callValue": "0",
+    "ethereum.evm.callState.wordStack": wordstack,
+    "ethereum.evm.callState.localMem": act.internal ? "_" : ".Map => _",
+    "ethereum.evm.callState.pc": pc,
+    "ethereum.evm.callState.gas": gas,
+    "ethereum.evm.callState.memoryUsed": act.internal ? "VMemoryUsed" : "0 => _",
+    "ethereum.evm.callState.callDepth": "VCallDepth => _",
+    "ethereum.evm.output":             output,
+    "ethereum.evm.statusCode":         "_ => " + statusCode,
+    "ethereum.network.activeAccounts": activeAccounts,
+    "ethereum.network.accounts":       accounts,
+  })
+    const buildAccount = varname => {
+      let fstore = Object.keys(act.storage[varname])
+      .map(key => storageEntry(varname, key, act.storage[varname][key], pass && !oog))
+      .concat(["_:Map"])
+      .reverse()
+      .reduceRight(brackedJoin(STORAGE_DELIMITER), "")
+
+    if(!(act.varname2alias[varname])) warn(`Implementation of variable "${varname} : address" not found!`);
+    const alias = act.varname2alias[varname];
+    if(alias && !(config.implementations[alias])) warn(`Implementation of contract alias ${alias} not found!`);
+    const contract_name = alias
+      && alias in config.implementations
+      && config.implementations[alias].name;
+    const balance = act.balance
+      && act.balance[varname]
+      || varname + "_balance";
+
+    return {account: {
+      acctID:  varname,
+      balance: balance,
+      code:    `#parseByteStack("0x${contract_name && config.contracts[contract_name].bin_runtime || ""}")`,
+      storage: fstore,
+      nonce:   "_"
+    }}
+  }
 
-const buildAct = config => act => {
-  console.log("Building k-spec " + act.subject + "." + act.name)
+
+  // TODO - simplify this
   if(!act.storage) act.storage = {};
   if(!act.storage.ACCT_ID) act.storage.ACCT_ID = []
+  const storage = {
+    ...act.storage,
+    [act.subject]: act.storage.ACCT_ID
+  }
+  delete storage.ACCT_ID;
+
+  // GAS
+  const gas = "VGas => " + ((act.hasGas && !oog) ? "VGas -Int " + act.gas : "_");
```

Comment:
The post gas should not be the cleaned up gas expression, but rather the "raw" expression, just tokenized.

Reply:
:+1: